src/data/local_storage.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * @module Data
+ * @submodule LocalStorage
+ * @requires core
+ *
+ * This module defines the p5 methods for working with local storage
+ */
+
+'use strict';
+var p5 = require('../core/main');
+/**
+ *
+ * Stores a value in local storage under the key name. The key can
+ * be the name of the variable but it does not have to be. To retrieve
+ * stored items see <a href="#/p5/loadItem">loadItem</a>.
+ *
+ * @method storeItem
+ * @for p5
+ * @param {String} key
+ * @param {String|Number|Object|Boolean|p5.Color} value
+ *
+ * @example
+ * <div><code>
+ * // Type to change the letter in the
+ * // center of the canvas.
+ * // If you reload the page, it will
+ * // still display the last key you entered
+ *
+ * let myText;
+ *
+ * function setup() {
+ *   createCanvas(100, 100);
+ *   myText = loadItem('myText');
+ *   if (myText === null) {
+ *     myText = '';
+ *   }
+ * }
+ *
+ * function draw() {
+ *   textSize(40);
+ *   background(255);
+ *   text(myText, width / 2, height / 2);
+ * }
+ *
+ * function keyPressed() {
+ *   myText = key;
+ *   storeItem('myText', myText);
+ * }
+ * </code></div>
+ *
+ * @alt
+ * When you type the key name is displayed as black text on white background.
+ * If you reload the page, the last letter typed is still displaying.
+ *
+ */
+p5.prototype.storeItem = function(key, value) {
+  if (value === 'undefined') {
+    console.log('You cannot store undefined variables using storeItem()');
```

Comment:
you might be already thinking this, but i think it would be awesome to use the friendly error system for these messages!

Reply:
I do think this could be nice. I am decidedly uncomfortable working with the FES code, but I should do it anyways. I should at least check the parameters with FES. I find that when I am logging a single very specific message that using the FES actually makes things much harder. But in that situation I guess the benefit is that people can turn it off?